### PR TITLE
[improve][cleanup] Keep usage of deprecated configs inside ServiceConfiguration

### PR DESCRIPTION
**Pull request by tisonkun**
### Motivation

Cleanup code when reading it.

### Modifications

Encapsulate usage of deprecated configs:

* subscriptionKeySharedEnable
* zookeeperServers

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

- [x] `doc` 

Configuration document may change, reflecting deprecations.

### CODE REVIEW

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java [hunk 1]
```diff
@@ -2779,6 +2781,10 @@ public class ServiceConfiguration implements PulsarConfiguration {
     )
     private Set<String> additionalServlets = new TreeSet<>();
 
+    public boolean isSubscriptionKeySharedEnable() {
+        return subscriptionKeySharedEnable && subscriptionTypesEnabled.contains("Key_Shared");
+    }
```
Comment (codelipenghui) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java:2786]: Here should be a bug fix. If the subscriptionKeySharedEnable = true but the subscriptionTypesEnabled doesn't have the `Key_Shared` subscription, we should treat it as Key_Shared subscription disabled.

https://github.com/apache/pulsar/blob/b68415a89dcd067511d01956447d1a507865534c/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java#L704-L708

Could you please add a UT?

Comment (tisonkun) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java:2786]: I'm glad to do so. Shall I separate the bug fix into a new PR or join in this PR?

Comment (tisonkun) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java:2786]: @codelipenghui I'll push a commit to add a UT along with this patch later. Please confirm whether it's proper when you have spare time :)

Comment (tisonkun) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java:2803]: Updated. Please give it another pass of review.

cc @codelipenghui

**Comment by tisonkun** [comment]
cc @hangc0276

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java [hunk 2]
```diff
@@ -2839,15 +2849,8 @@ public String getBookkeeperMetadataStoreUrl() {
...
-            if (StringUtils.isNotBlank(metadataStoreUrl)) {
-                suffix = metadataStoreUrl;
-            } else {
-                // Fallback to old setting
-                // Note: chroot is not settable by using 'zookeeperServers' config.
-                suffix = ZKMetadataStore.ZK_SCHEME_IDENTIFIER + zookeeperServers;
-            }
-            return "metadata-store:" + suffix + BookKeeperConstants.DEFAULT_ZK_LEDGERS_ROOT_PATH;
+            // Note: chroot is not settable by using 'zookeeperServers' config.
+            return "metadata-store:" + getMetadataStoreUrl() + BookKeeperConstants.DEFAULT_ZK_LEDGERS_ROOT_PATH;
```
[the remark this answers is not in the document]
Comment (tisonkun) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java:2853]: The configuration will resolve empty scheme as ZK. Once we decide to break backward compatibility, we should remove all usage of `zookeeperServers` and this fallback logic will be remove also. Thus, I don't think we should explicit specify the identifier, or we should explicit add it in `getMetadataStoreUrl` instead.

**Comment by tisonkun** [comment]
Weird. Many test cases failed due to container failure.

**Comment by tisonkun** [comment]
Hit flaky test. Reported at https://github.com/apache/pulsar/issues/16558.

**Comment by tisonkun** [comment]
@hangc0276 @codelipenghui All required tests are passed. Please take a look.

**Comment by tisonkun** [comment]
ping @codelipenghui @hangc0276

**Review by codelipenghui — APPROVED** [review]

**Review by hangc0276 — APPROVED** [review]

**Pull request merged**